src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java

```diff
@@ -117,36 +218,184 @@ public void visitToken(DetailAST ast) {
+     *
+     * @param token expected to be TokenType.METHOD_DEF .
+     * @return modifier string for {@code token} with 'package' scope added if needed
+     */
+    private static String extractModifiersString(final DetailAST token) {
+        final List<String> collectedModifiers = getModifiers(token);
+        final boolean isInsideInterface =
+            token.getParent().getParent().getType() == TokenTypes.INTERFACE_DEF;
+
+        if (Collections.disjoint(SCOPE_MODIFIERS, collectedModifiers)) {
+            String defaultModifier = PACKAGE_MODIFIER;
+            if (isInsideInterface) {
+                defaultModifier = PUBLIC_MODIFIER;
+            }
+            collectedModifiers.add(0, defaultModifier);
+        }
+
+        // methods in interfaces can only have public, abstract, default, private,
+        // static modifiers. abstract is only valid when used with just the public
+        // modifier, so simply add it to the end of the list if it is missing
+        if (isInsideInterface && missingAbstractModifier(collectedModifiers)) {
+            collectedModifiers.add(ABSTRACT_MODIFIER);
+        }
+
+        return String.join(" ", collectedModifiers);
+    }
+
+    /**
+     * Get the modifiers associated with a method.
+     *
+     * @param token the method token
+     * @return a list of modifiers found on the method, never empty
+     */
+    private static List<String> getModifiers(final DetailAST token) {
+        final List<String> collectedModifiers = new LinkedList<>();
+        final DetailAST modifiersAst = token.findFirstToken(TokenTypes.MODIFIERS);
+        DetailAST modifier = modifiersAst.getFirstChild();
+        while (modifier != null) {
+            // skipping annotations
+            if (modifier.getType() != TokenTypes.ANNOTATION) {
+                collectedModifiers.add(modifier.getText());
+            }
+            // advance element in loop
+            modifier = modifier.getNextSibling();
+        }
+        return collectedModifiers;
+    }
+
+    /**
+     * Determine if the {@code abstract} modifier is missing from the list of modifiers. This
+     * method is only assumed to be called on a collection of modifiers describing a method
+     * within an interface. {@code abstract} is considered missing if the current list of modifiers
+     * has only one element, and that element is the literal string {@code public}. Based on the
+     * <a href="https://docs.oracle.com/javase/specs/jls/se9/html/jls-9.html#jls-9.4">JLS</a>
+     * this is the only valid combination use of the {@code abstract} modifier when inside of
+     * an interface.
+     *
+     * @param modifiers the list of modifiers for the method
+     * @return {@code true} if {@code abstract} should be added to the modifier list
+     */
+    private static boolean missingAbstractModifier(final List<String> modifiers) {
+        return modifiers.size() == 1 && PUBLIC_MODIFIER.equals(modifiers.get(0));
```

Comment:
I did it this way because it's faster to check the size and explicit existence of `public` as the first item than to check all items to see if they equal to `static`, `default`, or `private`.  Since this only applies to the modifiers on methods, If a method inside of an interface is declared `public static` then it can never be abstract